plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerManager.kt

```diff
@@ -172,22 +175,32 @@ class CodeModernizerManager(private val project: Project) : PersistentStateCompo
             }
             val validatedBuildFiles = project.getSupportedBuildFilesWithSupportedJdk(supportedBuildFileNames, supportedJavaMappings)
             return if (validatedBuildFiles.isNotEmpty()) {
-                ValidationResult(true, validatedBuildFiles = validatedBuildFiles, validatedProjectJdkName = projectJdk?.description.orEmpty())
+                ValidationResult(
+                    true,
+                    validatedBuildFiles = validatedBuildFiles,
+                    validatedProjectJdkName = projectJdk?.description.orEmpty(),
+                    buildSystem = CodeTransformBuildSystem.Maven,
+                    buildSystemVersion = getMavenVersion(project)
+                )
             } else {
                 ValidationResult(
                     false,
-                    message("codemodernizer.notification.warn.invalid_project.description.reason.no_valid_files", supportedBuildFileNames.joinToString()),
-                    InvalidTelemetryReason(
+                    invalidReason = message(
+                        "codemodernizer.notification.warn.invalid_project.description.reason.no_valid_files",
+                        supportedBuildFileNames.joinToString()
+                    ),
+                    invalidTelemetryReason = InvalidTelemetryReason(
                         CodeTransformPreValidationError.NonMavenProject,
                         if (isGradleProject(project)) "Gradle build" else "other build"
-                    )
+                    ),
+                    buildSystem = if (isGradleProject(project)) CodeTransformBuildSystem.Gradle else CodeTransformBuildSystem.Unknown
```

Comment:
Right now this `isGradleProject()` is just used for telemetry right? Looks like it's implemented here: https://github.com/aws/aws-toolkit-jetbrains/blob/3c8beb6ddf411daf1131d9525733c644a95985c3/plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/utils/CodeTransformValidationUtils.kt#L10

Reply:
yes! It's just for telemetry and it's using the function from the util.